src/pyodide/internal/workers-api/src/workers/_workers.py

```diff
@@ -1114,28 +1114,53 @@ def __init__(self, js_step):
         self._js_step = js_step
         self._memoized_dependencies = {}
         self._in_flight = {}
+        self.step_closures = {}
 
     def do(self, name, depends=None, concurrent=False, config=None):
         def decorator(func):
             async def wrapper():
+                # if implicit params are enabled, each param that is not context should be treated as a dependency and resolved
+                # In other words, we introspect the declaration and call a function (need to make sure it's a step) with the same name as the corresponding param
+                # This new code path should discard depends, as we encourage users to implicitly declare their invariant steps in the signature
+                # if the compat flag is disabled, then we just maintain the same legacy behavior
```

Comment:
If we're going to ignore the `depends` argument, I think we should raise if it isn't `None`.